### PR TITLE
Fixes VSTS Bug 675488: [Feedback] Changing tab doesn't allow to

### PR DESCRIPTION
**Pull request by mkrueger**
directly type

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/675488

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/SdiWorkspaceWindow.cs [hunk 1]
```diff
@@ -287,10 +288,14 @@ public void SelectWindow()
 			// The tab change must be done now to ensure that the content is created
 			// before exiting this method.
 			tabControl.CurrentTabIndex = tab.Index;
-
 			// Focus the tab in the next iteration since presenting the window may take some time
 			Application.Invoke ((o, args) => {
 				DockNotebook.ActiveNotebook = tabControl;
+				var editor = ActiveViewContent.GetContent<Ide.Editor.TextEditor> ();
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/SdiWorkspaceWindow.cs]: Should this know of the editor? Shoudln't we do that on ActiveWindowChanged somewhere else?

**Comment by mkrueger** [comment]
Y it still repros.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/SdiWorkspaceWindow.cs (hunk 1 above)
Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/SdiWorkspaceWindow.cs]: Editor should be in focus if there is an editor. It's a bit more complex for other controls. The proper thing would be that the workspace window stores the widget that had the focus on focus lost and give it back in the OnSelect event.

However I wasn't able to find a good implementation for it. Esp. the search widget is generated after the workspace window initialization and the gtk focus APIs are a bit limited. The workspace window atm controls the focusing on selection that's why it's there. I don't want to clutter that logic up too much.

**Comment by mkrueger** [comment]
@monojenkins backport release-7.8

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/SdiWorkspaceWindow.cs (hunk 1 above)
Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/SdiWorkspaceWindow.cs]: @slluis ^^

Comment (slluis) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/SdiWorkspaceWindow.cs]: Marius is right, we should not special case the editor here.

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/SdiWorkspaceWindow.cs]: Any suggestions ? DeepGrabFocus needs to be bypassed in that case.

**Comment by mkrueger** [comment]
@monojenkins backport release-7.8

**Comment by mkrueger** [comment]
Moved the logic into the view content. But that means that every view content needs to handle it itself correctly.

**Comment by Therzok** [comment]
This might conflict with @slluis work to rework the document model.

**Pull request merged**